price-feeder/router/v1/response.go

```diff
@@ -24,3 +27,22 @@ type (
 		Prices map[string]sdk.Dec `json:"prices"`
 	}
 )
+
+// errorResponse defines the attributes of a JSON error response.
+type errorResponse struct {
+	Code  int    `json:"code,omitempty"`
+	Error string `json:"error"`
+}
+
+// newErrorResponse creates a new errorResponse instance.
+func newErrorResponse(code int, err string) errorResponse {
+	return errorResponse{Code: code, Error: err}
+}
+
+// writeErrorResponse prepares and writes a HTTP error
+// given a status code and an error message.
+func writeErrorResponse(w http.ResponseWriter, status int, err string) {
+	w.Header().Set("Content-Type", "application/json")
+	w.WriteHeader(status)
+	_, _ = w.Write(legacy.Cdc.MustMarshalJSON(newErrorResponse(0, err)))
```

Comment:
No need for this dependency. Let's just use `json` directly.

```suggestion
  bz, _ := json.Marshal(newErrorResponse(0, err))
	_, _ = w.Write(bz)
```